src/systems/visual/entity.ts

```diff
@@ -78,18 +96,36 @@ export class NodeEntity extends Entity {
     });
   }
 
-  static createNode(x: number, y: number, color: string): konva.Circle {
-    let node = new konva.Circle({
+  static createNode(
+    x: number,
+    y: number,
+    color: string,
+    key?: string
+  ): [konva.Group, konva.Shape] {
+    let group = new konva.Group({
       x: x,
       y: y,
+      draggable: true
+    });
+
+    let node = new konva.Circle({
+      x: 0,
+      y: 0,
       radius: 10,
       fill: color,
       stroke: "black",
       strokeWidth: 2,
-      draggable: true
+      draggable: key === undefined,
```

Comment:
since the group as a whole is draggable, there's no need for the circle to be draggable ever.
```suggestion
```